REFACTOR - HISTORY - Use case base impl

Deivid117 opened the pull request:
- Se implementó el caso de uso base para todos los casos de uso del historial.
- Se modificó el query para obtener el listado de conversiones (obtiene solo los tipos de cambio que contengan al menos una conversión asociada).

[Tareas](https://trello.com/c/qKsIDEqV)

Deivid117 committed:
REFACTOR - Use case base impl

- Se implementó el caso de uso base para obtener, eliminar, actualizar y obtener favoritos de las conversiones.
- Se modificó el query de obtener el listado de conversiones para que obtenga solo los tipos de cambio que tenga al menos una conversion asociada.

Deivid117 committed:
Merge branch 'feature/delete-update-conversion' of https://github.com/Deivid117/DolarBlueApp into refactor/use-case-base-impl

Deivid117 committed:
REFACTOR - Use case base impl for Search conversion use case

- Se implemetó el caso de uso base para la búsqueda de conversiones.

- Se modificó el query para obtener la lista de conversiones sin el uso de flow y que devuelva solo tipos de cambio que tengan al menos una conversión asociada.

- Se eliminó código comentado de los casos de uso

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into refactor/use-case-base-impl

# Conflicts:
#	app/src/main/java/com/varqulabs/dolarblue/history/domain/useCases/SearchConversionsHistoryUseCase.kt

Deivid117 committed:
REFACTOR - Use case base impl for Search conversions

- Se agregó el caso de uso base para el caso de uso de búsqueda de conversiones

juanjosevarqu reviewed (approved):
Nitido

Deivid117 committed:
FEATURE - Conversion counter and delete current exchange

- Se agregaron los dao para obtener cuántas conversiones asociadas tiene un tipo de cambio y para eliminar un tipo de cambio cuando no tengan ninguna conversión.

- Se agregaron los repository y use case para contar las conversiones de un tipo de cambio y eliminar un tipo de cambio.

- Se modificó el query de obtener el listado de conversiones de forma más simplificada.

Deivid117 committed:
Merge branch 'develop' of https://github.com/VarquLabs/DolarBlueApp into refactor/use-case-base-impl

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/repository/ConversionsHistoryRepository.kt:
```diff
@@ -13,6 +13,10 @@ interface ConversionsHistoryRepository {
 
+    suspend fun getTheExchangeRateConversionCount(id: Int): Int
+
+    suspend fun deleteCurrentExchangeRate(currentExchangeId: Int)
```
es necesario el prefijo current ?

Deivid117 replied:
No realmente, era por usar el mismo nombre de la tabla, pero sin el prefijo se lee mejor

juanjosevarqu commented on app/src/main/java/com/varqulabs/dolarblue/history/domain/repository/ConversionsHistoryRepository.kt:
```diff
@@ -13,6 +13,10 @@ interface ConversionsHistoryRepository {
 
     suspend fun updateConversion(conversion: Conversion)
 
+    suspend fun getTheExchangeRateConversionCount(id: Int): Int
```
el "The" queda un poco raro

Deivid117 committed:
REFACTOR - Functions renamed

- Se renombraron funciones, casos de uso y parámetros por unos más descriptivos.

juanjosevarqu reviewed (approved):
go
